src/daft-local-execution/src/sinks/outer_hash_join_probe.rs

```diff
@@ -0,0 +1,417 @@
+use std::sync::Arc;
+
+use arrow2::bitmap::MutableBitmap;
+use common_error::DaftResult;
+use daft_core::{
+    prelude::{Schema, SchemaRef},
+    series::Series,
+};
+use daft_dsl::ExprRef;
+use daft_micropartition::MicroPartition;
+use daft_plan::JoinType;
+use daft_table::{GrowableTable, Probeable, Table};
+use indexmap::IndexSet;
+use tracing::{info_span, instrument};
+
+use super::streaming_sink::{StreamingSink, StreamingSinkOutput, StreamingSinkState};
+use crate::pipeline::PipelineResultType;
+
+struct IndexBitmapTracker {
```

Comment:
I think it make sense to have this as `IndexBitmapBuilder` which holds onto the `MutableBitmaps` and then when you `.build()` it you get an `IndexBitmap` which can be `or`d or iterated on.